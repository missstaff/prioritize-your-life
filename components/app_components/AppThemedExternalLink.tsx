import { Href, Link } from "expo-router";
import { Platform } from "react-native";
import { openBrowserAsync } from "expo-web-browser";
import { type ComponentProps } from "react";
import { useThemeColor } from "@/hooks/useThemeColor";
import { COLORTHEME } from "@/constants/Colors";

type Props = Omit<ComponentProps<typeof Link>, "href"> & {
  href: Href<string | object>;
};

const AppThemedExternalLink = ({ href, style, ...rest }: Props) => {
  const color = useThemeColor(
    { light: COLORTHEME.light.background, dark: COLORTHEME.dark.background },
    "background"
  );
  return (
    <Link
      target="_blank"
      {...rest}
      href={href}
      onPress={async (event) => {
        if (Platform.OS !== "web") {
          // Prevent the default behavior of linking to the default browser on native.
          event.preventDefault();
          // Open the link in an in-app browser.
          await openBrowserAsync(href as string);
        }
      }}
      style={[{ color }, style]}
    />
  );
};

export default AppThemedExternalLink;
