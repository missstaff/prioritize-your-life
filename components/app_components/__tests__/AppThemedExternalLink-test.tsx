import { render } from "@testing-library/react-native";
import { AppThemedExternalLink } from "../AppThemedExternalLink";

jest.mock("expo-web-browser", () => ({
  openBrowserAsync: jest.fn(),
}));

describe("AppThemedExternalLink", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const mockHref = "https://example.com";
  const mockStyle = { fontSize: 16 };

  it("matches snapshot", () => {
    const { toJSON } = render(
      <AppThemedExternalLink href={mockHref} style={mockStyle} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders the external link component", () => {
    const { getByTestId } = render(
      <AppThemedExternalLink href={mockHref} style={mockStyle} />
    );
    const externalLink = getByTestId("external-link");
    expect(externalLink).toBeDefined();
  });
});
