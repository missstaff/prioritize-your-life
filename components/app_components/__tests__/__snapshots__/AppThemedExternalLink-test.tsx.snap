// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppThemedExternalLink matches snapshot 1`] = `
<View
  testID="external-link"
/>
`;
